app/controllers/api/mobile/petitions_controller.rb

```diff
@@ -0,0 +1,26 @@
+class Api::Mobile::PetitionsController < ActionController::Base
+
+  attr_reader :mobile_service
+
+  def initialize(mobile_service: MobileApiService.new)
+    @mobile_service = mobile_service
+  end
+
+  def show
+    petition_id = params[:id]
+
+    petition_info = Rails.cache.fetch("mobile_petition_info:#{petition_id}", expires_in: cache_time) do
```

Comment:
We talked about doing this on the background, not during the user request. Should just fetch from the cache.